### PR TITLE
Json format

### PR DESCRIPTION
**Pull request by alseddnm**
Hello, I have removed my previous fork and created a new one.
Again, it is our 1st attempt to make a major contribution in Leshan, let us know if our code does not meet your conventions and standards, we are very flexible to make any modifications.

Thanks,
Nizar...!

### CODE REVIEW

**Comment by alseddnm** [comment]
@jvermillard Let us know if you have any feedback about this pull request?
I thought is much cleaner and easier remove the previous fork and created a new a one.
Thank You.

File: leshan-core/pom.xml [hunk 1]
```diff
@@ -38,7 +38,11 @@ Contributors:
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
         </dependency>
-
+		<dependency>
```
Comment (jvermillard) [inline leshan-core/pom.xml]: formatting broken

File: leshan-core/src/main/java/org/eclipse/leshan/core/model/ResourceModel.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public boolean isExecutable() {
     }
 
     public enum Type {
-        STRING, INTEGER, FLOAT, BOOLEAN, OPAQUE, TIME
+        STRING, INTEGER, FLOAT, BOOLEAN, OPAQUE, TIME, OBJECTLINK
```
Comment (jvermillard) [inline leshan-core/src/main/java/org/eclipse/leshan/core/model/ResourceModel.java]: OBJECTLINK must be a second PR

File: leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/LwM2mNodeDecoder.java [hunk 3]
```diff
@@ -111,6 +122,13 @@ public static LwM2mNode decode(byte[] content, ContentFormat format, LwM2mPath p
             }
             return new LwM2mResource(path.getResourceId(), Value.newBinaryValue(content));
         case JSON:
+        	try {
```
Comment (jvermillard) [inline leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/LwM2mNodeDecoder.java]: formatting (tabs??)

File: leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/LwM2mNodeDecoder.java [hunk 4]
```diff
@@ -228,6 +246,81 @@ private static LwM2mNode parseTlv(Tlv[] tlvs, LwM2mPath path, LwM2mModel model)
...
+        } else {
+            // resource level request
+        	Map<Integer,LwM2mResource> resourceMap = parseJsonPayLoadLwM2mResources(jsonObject,  path,  model);
+        	LwM2mResource resource = resourceMap.get(0);
+        	return resource;
+        }
+       
+    }
+    
+    public static Map<Integer,LwM2mResource> parseJsonPayLoadLwM2mResources(LwM2mJsonObject jsonObject, LwM2mPath path, LwM2mModel model) throws InvalidValueException
```
Comment (jvermillard) [inline leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/LwM2mNodeDecoder.java]: private

File: leshan-core/src/main/java/org/eclipse/leshan/json/JsonArrayElement.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+ *     Gemalto M2M GmbH
+ *******************************************************************************/
+
+package org.eclipse.leshan.json;
+
+import com.google.gson.annotations.SerializedName;
+
+public class JsonArrayElement{
+
+	@SerializedName("n")
```
Comment (jvermillard) [inline leshan-core/src/main/java/org/eclipse/leshan/json/JsonArrayElement.java]: yet again formatting,

if you do what written in the README it should work: just "mvn eclipse:eclipse" and import the project.

**Comment by jvermillard** [comment]
I created a branch and a new pull request, so I can fix stuff myself, please use this branch (json) if you whish to push more code.

**Comment by alseddnm** [comment]
Great.. Thanks Julien.

On Tue, Jun 16, 2015 at 5:25 AM, Julien Vermillard <notifications@github.com

**Pull request closed without merging**